interface/javascript/general.js

```diff
@@ -1,207 +1,245 @@
-		stageBorderBottom:0.06, 
-		logoPosX:0.06, 
-		logoPosY:0.06, 
-		logoHeight:0.07
-	}
-	);
+/*
+- unifier le listener resize
+- fair index.html avec redirect
+- faire check du poids du tout.
+- js hint
+- checl plus de ""
+
+- check dist sur daviddarx si probleme de load listener des images pour scene1, commne sur netlify
+- check dist sur daviddarx: est-ce que le logo blink au pageload? si oui, reessayer avec ancienne methode? 
+- window.location.hash=this.getAttribute("href").split("#")[1];
+- location.reload();
+*/
+
+//import "./add_stats.js";
+
+let i;
+
+const globalSettings = {
+	urlBase : 'stage-', 
+	animationsIDs : ['animation-01', 'animation-02', 'animation-03'], 
+	breakPointsDefinitions : [
+		{
+			width:1440, 
+			stageBorderLeft:0.05, 
+			stageBorderRight:0.05, 
+			stageBorderTop:0.05, 
+			stageBorderBottom:0.05, 
+			logoPosX:0.05, 
+			logoPosY:0.05, 
+			logoHeight:0.02
+		}, 
+		{
+			width:768, 
+			stageBorderLeft:0.05, 
+			stageBorderRight:0.05, 
+			stageBorderTop:0.05, 
+			stageBorderBottom:0.05, 
+			logoPosX:0.05, 
+			logoPosY:0.05, 
+			logoHeight:0.03
+		}, 
+		{
+			width:0, 
+			stageBorderLeft:0.06, 
+			stageBorderRight:0.06, 
+			stageBorderTop:0.06, 
+			stageBorderBottom:0.06, 
+			logoPosX:0.06, 
+			logoPosY:0.06, 
+			logoHeight:0.07
+		}
+	]
+};
+
+
+
+const Global = function(){
+	
+	this.url = undefined;
+	this.stageID = undefined;
+	
+	this.domRefs = { };
 	
-	window.windowSize = {
+	this.stageAnimationOutDuration = undefined;
+	
+	this.windowSize = {
 		width:undefined, 
 		height:undefined
-	};
-	
-	window.stageSettings = {
+	}, 
+	this.stageSettings = {
 		posX:undefined, 
 		posY:undefined,
 		width:undefined, 
-		height:undefined, 
-	};
-	
-	window.logoSettings = {
+		height:undefined
+	}, 
+	this.logoSettings = {
 		posX:undefined, 
 		posY:undefined,
 		width:undefined, 
 		height:undefined
-	};
-	
-	
-	
+	}
+
+	this.externalResizeListener = undefined;
 	
-	const hasClass = function(el, className) {
-		return el.classList ? el.classList.contains(className) : new RegExp('\\b'+ className+'\\b').test(el.className);
+	this.init = () => {
+		this.url = window.location.pathname;
+		this.stageID = parseInt(this.url.substring(this.url.lastIndexOf('/')+1).split('.html')[0].split(globalSettings.urlBase)[1]);
+		
+		this.domRefs.$logo = document.querySelector('.stage__logo');
+		this.domRefs.$logo.addEventListener('click', (e) => this.logoClickListener(e));
+		this.domRefs.$logo.addEventListener('touchstart', (e) => this.logoClickListener(e));
+		this.domRefs.$logoImg = Array.prototype.slice.call(this.domRefs.$logo.querySelectorAll('img'));
+		this.domRefs.$logoLink = this.domRefs.$logo.getElementsByTagName('a');
+		
+		this.domRefs.$stage = document.querySelector('.stage');
+		this.domRefs.$stageContainer = document.querySelector('.stage__container');
+		
+		this.domRefs.$pagination = document.querySelector('.pagination');
+		
+		for(i=0; i<globalSettings.animationsIDs.length; i++){
```

Comment:
since `animationsIDs` is an array you should just do `animationsIDs.forEach(uid => {})`